Spigot-Server-Patches/0436-Improved-Watchdog-Support.patch

```diff
@@ -257,14 +258,16 @@ index 557f80accfa36b495c9a8cffdab2e248c1cbb514..ec1f36736d79d4054ad7ff4da4e3659f
              if (dedicatedserverproperties.announcePlayerAchievements != null) {
                  ((GameRules.GameRuleBoolean) this.getGameRules().get(GameRules.ANNOUNCE_ADVANCEMENTS)).a(dedicatedserverproperties.announcePlayerAchievements, (MinecraftServer) this);
              }
-@@ -407,6 +407,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
+@@ -407,7 +407,8 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
              //this.remoteStatusListener.b(); // Paper - don't wait for remote connections
          }
  
+-        System.exit(0); // CraftBukkit
 +        hasFullyShutdown = true; // Paper
-         System.exit(0); // CraftBukkit
++        System.exit(this.abnormalExit ? -1 : 0); // CraftBukkit // Paper
```

Comment:
Are you sure this should be a negative number and not a small positive one? Is it to avoid clashes with exit codes in other places or what's the reasoning?

Reply:
a negative exit id seems really eer, am not sure anywhere where they're sanely used or what platform support for that looks like, generally you just return `1` as an exit code and call it
70 is apparently the defined "server incurred an error" shutdown status
`#define EX_SOFTWARE     70      /* internal software error */`

ref:
https://stackoverflow.com/questions/1101957/are-there-any-standard-exit-status-codes-in-linux
https://www.freebsd.org/cgi/man.cgi?query=sysexits&apropos=0&sektion=0&manpath=FreeBSD+4.3-RELEASE&format=html